### PR TITLE
Make tokens at the begining of a new line start on that line.

### LINKED ISSUE #128: Start position of tokens at the beginning of the line are recorded as being at the end of the previous line.
I was experimenting with using [nom_locate](https://github.com/fflorent/nom_locate/) for tracking source positions. I've got a mostly working branch with that, but I discovered that it generates different source positions in a number of places:
- any token at the beginning of the line currently has its position as being at the end of the previous line
- a whitespace token consisting of a single new line has start and end position with the same line and column, on the previous line

I think either decision for the second difference could be argued to be correct, but I think the first one is definitely an error. I've got a patch to make the first change (and am working on updating the tests), but I'm wondering if the second change also wants to be made.

### PR DESCRIPTION
**Pull request by tomprince**
Fixes #128.

### CODE REVIEW

**Comment by Kampfkarren** [comment]
Can you update the CHANGELOG as well, under an "Unreleased" category if one does not exist already?

CC @JohnnyMorganz, if you are interested or have any comments with regards to this.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 - Updated dependency cfg_if to v1.0
 
+### Fixed
+- Fixed the start position of tokens at the beginnig of a line to not be at the end of the previous line.
```
Comment (JohnnyMorganz) [inline CHANGELOG.md:15]:
```suggestion
- Fixed the start position of tokens at the beginning of a line to not be at the end of the previous line.
```

**Review by JohnnyMorganz — COMMENTED** [review]
Looks good to me, I have nothing to add

**Comment by Kampfkarren** [comment]
Thanks!

**Pull request merged**